parser/src/dataset/read.rs

```diff
@@ -326,6 +326,18 @@ where
 
                     Some(Ok(DataToken::SequenceStart { tag, len }))
                 }
+                Ok(DataElementHeader {
+                    tag,
+                    vr: VR::UN,
```

Comment:
Just checked the relevant section of the standard recently ([part 5, 7.1.2](http://dicom.nema.org/medical/dicom/current/output/chtml/part05/chapter_7.html#sect_7.1.2)), it is actually possible for elements of other VRs to have an undefined length. The full list is: OB, OD, OF, OL, OV, OW, SQ, and UN. The ones which are still missing and would break the parser are OW, OL, OV, OF, and OD.

I should check the implications of supporting these value representations.

Reply:
The decision was to treat such elements as data set sequences regardless of the VR. Should exceptions to this rule come around, we can re-evaluate it, but nevertheless this would be an improvement to the previous algorithm.